chore: add sourcemaps for JS and minified files #10999

ncoden opened the pull request:
#### Changes:
* add sourcemaps for js files (`foundation.js` and all plugins)
* add sourcemaps for minified files (`foundation.min.js`, `foundation-*.min.css`)
* improve documentation for `dist:javascript` gulp task

Related to https://github.com/zurb/foundation-sites/pull/10998 (@DanielRuf)
Closes https://github.com/zurb/foundation-sites/issues/10999 (@ncoden)

ncoden committed:
chore: add JS sourcemaps

See https://github.com/shama/webpack-stream#source-maps

ncoden committed:
chore: add sourcemaps for minified files

ncoden committed:
style: add some doc for minified source map

DanielRuf commented:
Will check tomorrow.

ncoden commented:
> Will check tomorrow.

Poke @DanielRuf 😄

DanielRuf commented on gulp/tasks/deploy.js:
```diff
@@ -44,28 +45,51 @@ gulp.task('deploy:version', function() {
       .pipe(rename({ suffix: '.min' }))
+      .pipe(cleancss({ compatibility: 'ie9' }))
+      .pipe(sourcemaps.write('.', {
+        mapFile: function(path) { return path.replace('.css.map', '.min.css.map'); }
```
This generates `.min.min.map`, should be probably `.min.map`.

ncoden replied:
No because `Sourcemaps are initialized before the ".min" renaming to be able retrieve original sourcemaps`. So original sourcemaps `xxx.css.map` are found because files are still like `xxx.css`, and this generate `xxx.css.map` sourcemaps after the minification we rename to `xxx.min.css.map`.

DanielRuf replied:
![image](https://user-images.githubusercontent.com/827205/37626257-89ebfcd0-2bcf-11e8-9218-421fe1907886.png)

![image](https://user-images.githubusercontent.com/827205/37626237-70683080-2bcf-11e8-963a-1182a1f1de2a.png)

DanielRuf replied:
Should be

```js
    .pipe(cssFilter)
      .pipe(gulp.dest('./dist/css'))
      .pipe(sourcemaps.init({ loadMaps: true }))
      .pipe(rename({ suffix: '.min' }))
      .pipe(cleancss({ compatibility: 'ie9' }))
      .pipe(sourcemaps.write('.'))
      .pipe(gulp.dest('./dist/css'))
      .pipe(cssFilter.restore)

    .pipe(jsFilter)
      .pipe(gulp.dest('./dist/js'))
      .pipe(sourcemaps.init({ loadMaps: true }))
      .pipe(rename({ suffix: '.min' }))
      .pipe(uglify())
      .pipe(sourcemaps.write('.'))
      .pipe(gulp.dest('./dist/js'));
```

DanielRuf replied:
Where there are written does only matter, the initialization collects the needed files. ;-)

ncoden replied:
hmm... this worked on my side. I'll check that again. Thanks for the review ;)

DanielRuf replied:
https://github.com/gulp-sourcemaps/gulp-sourcemaps/blob/master/src/init/index.js#L134
https://github.com/gulp-sourcemaps/gulp-sourcemaps/blob/master/src/write/index.internals.js#L106

ncoden committed:
fix: remove useless manual sourcemaps prefixing in gulp "deploy:dist" task

Sourcemaps output names are resolved at writting once source files are already renamed with the `.min` prefix.

ncoden committed:
Merge branch 'develop' into chore/and-js-min-sourcefiles-10999

ncoden commented:
@DanielRuf Fixed.
